lib/gis/parser_json.c

```diff
@@ -340,19 +340,27 @@ char *G__json(void)
  */
 void check_create_import_opts(struct Option *opt, char *element, FILE *fp)
 {
-    int i = 0;
+    int i = 0, j = 0;
     int has_import = 0;
-    char **tokens;
+    char **tokens, **urltokens;
 
+    G_debug(3,"tokenize string: <%s>", opt->answer);
     tokens = G_tokenize(opt->answer, "@");
     while (tokens[i]) {
         G_chop(tokens[i]);
         i++;
     }
 
+    /* check if it contains an URL, if ":" present, then URL, map@mapset otherwise */
+    urltokens = G_tokenize(tokens[1], ":");
+    while (urltokens[j]) {
+        G_chop(urltokens[j]);
+        j++;
+    }
```

Comment:
Why not simple `strchr()` for char in string check?

Reply:
A simple `strchr()` would be ok if more than one ":" is allowed, then split at the first occurrence of ":". Otherwise, as above, check if there are more than two tokens, if yes, fatal error.

Reply:
Thanks for the fatal error idea, added in b601c83

Reply:
So no port number in the URL? Colon can also be part of URL in general, e.g., https://grasswiki.osgeo.org/wiki/Category:Documentation. Back to my other comment: Regular expressions would be quite appropriate for JSON/web use case here, but some manual check for a sequence of characters would likely do the trick here as well.

Reply:
OK, entirely rewritten in 19d8065, now using `strstr()` instead of the tokenizer.